docs/direnv.md

```diff
@@ -0,0 +1,146 @@
+if ! has nix_direnv_version || ! nix_direnv_version 2.3.0; then
+  source_url "https://raw.githubusercontent.com/nix-community/nix-direnv/2.3.0/direnvrc" "sha256-Dmd+j63L84wuzgyjITIfSxSD57Tx7v51DMxVZOsiUD8="
+fi
+# https://github.com/input-output-hk/devx Slightly opinionated shared GitHub Action for Cardano-Haskell projects 
+use flake "github:input-output-hk/devx#ghc8107-iog"
+```
+
+This will tell `direnv` to instantiate a shell with the same environment as `nix develop github:input-output-hk/devx#ghc8107-iog`, when changing into the folder where the `.envrc` resides.
+
+You can also pin the flake to a specific version. This is a good practice to ensure reproducibility across different machines and systems. Refer to the Nix Flakes documentation on how to do this. Then, allow Direnv to take effect:
+
+```bash
+$ direnv allow
+```
+
+`direnv` will use the specified flake to provide a shell with all the Haskell development tools you need, as configured in the Devx repository.
+
+## Setting up your editor
+
+This setup should work with any editor that has support for language severs and the ability to use direnv to instantiate the necessary environment (`PATH`, ...). We will focus on the three most popular ones right now: VSCode, Emacs and Vim.
+
+### Visual Studio Code (VSCode)
+
+First, ensure that you have the following extensions installed:
+
+- [Nix Extension Pack](https://marketplace.visualstudio.com/items?itemName=pinage404.nix-extension-pack)
+- [Haskell](https://marketplace.visualstudio.com/items?itemName=haskell.haskell)
+
+The Nix Extension Pack brings `direnv` support, while the Haskell extension provides support for the Haskell Language Server.
+
+When using `direnv` with VSCode, ensure that VSCode has the correct environment loaded. Restart VSCode or reload the window once the shell has been instantiated. If you're on a platform where launching VSCode from the terminal is supported, you can also launch VSCode with `code .` from the terminal within the `direnv` environment.
+
+### Emacs
+
+For Emacs, ensure you have the following packages installed:
+
+- [`lsp-mode`](https://github.com/emacs-lsp/lsp-mode)
+- [`lsp-haskell`](https://github.com/emacs-lsp/lsp-haskell)
+
+You can use the following config for `direnv` and `lsp-haskell`:
+
+```emacs-lisp
+;; direnv
+(use-package direnv
+  :config
+  (direnv-mode))
+
+;; Haskell Language Server
+(use-package lsp-mode
+  :commands lsp
+  :config
+  (require 'lsp-haskell)
+  (add-hook 'haskell-mode-hook #'lsp))
+```
+
+Now, when you open a Haskell file in your project, `direnv` should automatically load your environment, and lsp-haskell should start providing IDE-like features.
+
+### Vim
+
+For Vim, we recommend the [`direnv.vim`](https://github.com/direnv/direnv.vim) and [`haskell-vim`](https://github.com/neovimhaskell/haskell-vim) plugins. You can install these plugins using a plugin manager like [`vim-plug`](https://github.com/junegunn/vim-plug).
```

Comment:
I would like to recommend users to rely on https://github.com/neoclide/coc.nvim. However, I'm aware that this may be seen as overly opinionated. WDYT?

Reply:
Mentioning it as an option is good. Try to be neutral in your assessment. We do not want to tell
People what to do but offer options they can choose from.